benchmark/_http-benchmarkers.js

```diff
@@ -0,0 +1,167 @@
+'use strict';
+
+const child_process = require('child_process');
+
+function AutocannonBenchmarker() {
+  this.name = 'autocannon';
+}
+
+AutocannonBenchmarker.prototype.autocannon_exe = process.platform === 'win32'
+                                               ? 'autocannon.cmd'
+                                               : 'autocannon';
+
+AutocannonBenchmarker.prototype.present = function() {
+  const result = child_process.spawnSync(this.autocannon_exe, ['-h']);
+  return !(result.error && result.error.code === 'ENOENT');
+};
+
+AutocannonBenchmarker.prototype.create = function(port, path, duration,
+                                                  connections) {
+  const args = ['-d', duration, '-c', connections, '-j', '-n',
+                `http://127.0.0.1:${port}${path}` ];
+  const child = child_process.spawn(this.autocannon_exe, args);
+  child.stdout.setEncoding('utf8');
+  return child;
+};
+
+AutocannonBenchmarker.prototype.processResults = function(output) {
+  let result;
+  try {
+    result = JSON.parse(output);
+  } catch (err) {
+    // Do nothing, let next line handle this
+  }
+  if (!result || !result.requests || !result.requests.average) {
+    return undefined;
+  } else {
+    return result.requests.average;
+  }
+};
+
+function WrkBenchmarker() {
+  this.name = 'wrk';
+  this.regexp = /Requests\/sec:[ \t]+([0-9\.]+)/;
+}
+
+WrkBenchmarker.prototype.present = function() {
+  const result = child_process.spawnSync('wrk', ['-h']);
+  return !(result.error && result.error.code === 'ENOENT');
+};
+
+WrkBenchmarker.prototype.create = function(port, path, duration, connections) {
+  const args = ['-d', duration, '-c', connections, '-t', 8,
+                `http://127.0.0.1:${port}${path}` ];
+  const child = child_process.spawn('wrk', args);
+  child.stdout.setEncoding('utf8');
+  child.stderr.pipe(process.stderr);
+  return child;
+};
+
+WrkBenchmarker.prototype.processResults = function(output) {
+  const match = output.match(this.regexp);
+  const result = match && +match[1];
+  if (!result) {
+    return undefined;
+  } else {
+    return result;
+  }
+};
+
+const http_benchmarkers = [ new WrkBenchmarker(),
+                            new AutocannonBenchmarker() ];
+
+const supported_http_benchmarkers = [];
+const installed_http_benchmarkers = [];
+var benchmarkers = {};
+
+http_benchmarkers.forEach((benchmarker) => {
+  const name = benchmarker.name;
+  const present = benchmarker.present();
+
+  benchmarkers[name] = {
```

Comment:
> I think having a separate array with name for all benchmarkers right after they are defined is more readable.

I'm not sure I understand exactly what you mean, but go ahead. My sugested implementation was just about removing one layer of object wrapping. 

> I can drop those, but IMHO those are nice to have. 
It is easy to add features later, but hard to remove them. I think either specifying the benchmarker(s) directly, or using the default is sufficient to cover most use cases. If it turns out that they are needed and have real use cases, I'm would be happy to accept them.